go/host/l1/publisher.go

```diff
@@ -239,6 +253,55 @@ func (p *Publisher) FetchLatestPeersList() ([]string, error) {
 	return filteredHostAddresses, nil
 }
 
+func (p *Publisher) GetImportantContracts() map[string]gethcommon.Address {
+	p.importantAddressesMutex.RLock()
+	defer p.importantAddressesMutex.RUnlock()
+	return p.importantContractAddresses
+}
+
+// ResyncImportantContracts will fetch the latest important contracts from the management contract and update the cached map
+// Note: this should be run in a goroutine as it makes L1 transactions in series and will block.
+// Cache is not overwritten until it completes.
+func (p *Publisher) ResyncImportantContracts() error {
```

Comment:
Should we rely on eth_call instead of fetching the data over the minted block txt ?